test-result-summary-client/src/PerfCompare/PerfCompare.jsx

```diff
@@ -469,11 +474,18 @@ export default class PerfCompare extends Component {
                 curTestScore = parsedVariantsTest.metrics[j].statValues.mean;
                 curTestCI = parsedVariantsTest.metrics[j].statValues.CI;
 
-                // Get the metric's units
-                // get BenchmarkRouter & Metric files from server
-                let parserProps = await getParserProps();
+                // To get the metric's higherbetter/units
+                // first check if Metric does exist in starte, if not get Metric info from server
+                let benchmark = curVariantData["benchmark"]
+                if( !this.metricsProps[benchmark])  {
+                    let propers = await getBenchmarkMetricProps(benchmark);
+                    if (propers) {
+                        this.metricsProps[benchmark] = propers;
+                    }
+                }
+                
+                let metricProps = this.metricsProps[benchmark][curMetricName];
```

Comment:
- If `propers` is empty, it means `this.metricsProps[benchmark]` is undefined. Line 487 will break. This logic should move to above if block.
- Please rename `propers ` to a meaningful name